lib/keystores.js

```diff
@@ -275,20 +274,10 @@ exports.getStorageUsage = async ({
     }
 
     // start counting keys in keystore
-    counters.push(_addKeyCount({usage, moduleApi, keystoreId, signal}));
-
-    // await counters if max concurrency reached
-    if(counters.length === USAGE_COUNTER_MAX_CONCURRENCY) {
-      await Promise.all(counters);
-      counters = [];
-    }
-
-    if(signal && signal.abort) {
-      throw new BedrockError(
-        'Computing metered storage aborted.',
-        'AbortError',
-        {meterId, httpStatusCode: 503, public: true});
-    }
+    counters.push(limit(() => {
+      _checkComputeStorageSignal({signal, meterId});
+      return _addKeyCount({usage, moduleApi, keystoreId, signal});
+    }));
```

Comment:
We still need to check the abort signal handler while processing configs.

```suggestion
    }));
    
    _checkComputeStorageSignal({signal, meterId});
```